src/Client/ClientBaseHelpers.cpp

```diff
@@ -131,6 +131,46 @@ std::string getChineseZodiac()
 }
 
 #if USE_REPLXX
+/// hasInvalidUTF8 checks if input string has invalid utf-8 sequences.
+static bool hasInvalidUTF8(const String & query)
+{
+    for (const char c : query)
+    {
+        /// Standalone utf  -8 continuation bytes (0x80-0xBF) cause counting inconsistency
+        if (static_cast<unsigned char>(c) >= 0x80 && static_cast<unsigned char>(c) <= 0xBF)
+            return true;
+    }
+    return false;
+}
+
+/// countCodePointsWithSeqLength counts utf-8 code points using iteration and `UTF8::seqLength`
+static size_t countCodePointsWithSeqLength(const char * begin, const char * end)
```

Comment:
Why not simply use `UTF8::seqLength` all the time? I doubt that we need performance in this place, highlighting has extra overhead anyway

Reply:
Sure, we can try. Except for the performance concerns no other issues. Fixed to remove the conditional check and use it all the time.